specification/datamigration/resource-manager/Microsoft.DataMigration/stable/2018-04-19/definitions/Common.json

```diff
@@ -0,0 +1,310 @@
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "ApiError": {
+      "type": "object",
+      "description": "Error information.",
+      "properties": {
+        "error": {
+          "description": "Error information in OData format",
+          "$ref": "#/definitions/ODataError"
+        }
+      }
+    },
+    "NameAvailabilityResponse": {
+      "type": "object",
+      "description": "Indicates whether a proposed resource name is available",
+      "properties": {
+        "nameAvailable": {
+          "type": "boolean",
+          "description": "If true, the name is valid and available. If false, 'reason' describes why not."
+        },
+        "reason": {
+          "type": "string",
+          "description": "The reason why the name is not available, if nameAvailable is false",
+          "enum": [
+            "AlreadyExists",
+            "Invalid"
+          ],
+          "x-ms-enum": {
+            "name": "NameCheckFailureReason",
+            "modelAsString": true
+          }
+        },
+        "message": {
+          "type": "string",
+          "description": "The localized reason why the name is not available, if nameAvailable is false"
+        }
+      }
+    },
+    "NameAvailabilityRequest": {
+      "type": "object",
+      "description": "A resource type and proposed name",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The proposed resource name"
+        },
+        "type": {
+          "type": "string",
+          "description": "The resource type chain (e.g. virtualMachines/extensions)"
+        }
+      }
+    },
+    "ODataError": {
+      "type": "object",
+      "description": "Error information in OData format.",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "The machine-readable description of the error, such as 'InvalidRequest' or 'InternalServerError'"
+        },
+        "message": {
+          "type": "string",
+          "description": "The human-readable description of the error"
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ODataError"
+          },
+          "description": "Inner errors that caused this error"
+        }
+      }
+    },
+    "Quota": {
+      "type": "object",
+      "description": "Describes a quota for or usage details about a resource",
+      "properties": {
+        "currentValue": {
+          "type": "number",
+          "format": "double",
+          "description": "The current value of the quota. If null or missing, the current value cannot be determined in the context of the request."
+        },
+        "id": {
+          "type": "string",
```

Comment:
Isn't this field read-only? if so should be marked as `"readOnly": true,`

Reply:
In general any property of a type that is supposed to be sent only in one direction from Server to Client should be marked as `"readOnly": true,`

---
In reply to: [183830418](https://github.com/Azure/azure-rest-api-specs/pull/2936#discussion_r183830418) [](ancestors = 183830418)